### PR TITLE
Beta release

### PR DESCRIPTION
**Pull request by peduarte**
🥁 https://github.com/modulz/stitches/pull/338

- Add migration guide alpha to beta
- Update docs to beta
- Update to latest DS
- Replace `next-mdx-enhanced` with `next-mdx-remote`
- Replace `react-live` with a simple `refractor` layer
- Add line highlight support
- Add word highlight support
- Add support for making word highlight clickable
- Add interaction between content and word highlight

Still to do:
- Add Nextjs `Image` component
- Finish updating docs
- Update RSS

### CODE REVIEW

File: pages/blog/migrating-from-alpha-to-beta.mdx [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+  }
+})
+```
+
+#### Choose your theme class
+
+You can now define the class to be generated when creating additional themes.
+
+```diff
++ const darkTheme = ('dark-theme', {})
```
Comment (christoph-fricke) [inline pages/blog/migrating-from-alpha-to-beta.mdx:197]: It looks like something is missing here, right? (I don't know the new api, just curious in the beta)
I'm guessing a `theme` function call? :D
```suggestion
+ const darkTheme = theme('dark-theme', {})
```

Comment (peduarte) [reply pages/blog/migrating-from-alpha-to-beta.mdx:197]: haha yeah 😂 glad its intuitive enough! haha

**Comment by joe-bell** [comment]
Would it be possible to expand the ["Server-side rendering"](https://stitches-site-git-beta.modulz-deploys.com/blog/migrating-from-alpha-to-beta#server-side-rendering) section with steps for **Next.js**?
Happy to make the addition myself but appreciate this is a draft PR and didn't want to make things more complicated!

From my findings so far it seems like the following works pretty nicely:

```tsx
// pages/_document.tsx
import * as React from "react";
import NextDocument, { DocumentContext } from "next/document";
import { getCssString } from "../stitches.config";

export default class Document extends NextDocument {
  static async getInitialProps(ctx: DocumentContext) {
    const originalRenderPage = ctx.renderPage;

    try {
      let extractedCss: string;

      ctx.renderPage = () => {
        const styles = getCssString();
        extractedCss = styles;

        return originalRenderPage();
      };

      const initialProps = await NextDocument.getInitialProps(ctx);

      return {
        ...initialProps,
        styles: (
          <>
            {initialProps.styles}
            <style dangerouslySetInnerHTML={{ __html: extractedCss }} />
          </>
        )
      };
    } finally {
    }
  }
}
```

**Comment by peduarte** [comment]
@joe-bell great idea. this is what i'd do:
```tsx
import React from 'react';
import NextDocument, { Html, Head, Main, NextScript } from 'next/document';
import { getCssString } from '../stitches.config';

export default class Document extends NextDocument {
  static async getInitialProps(ctx) {
    try {
      const initialProps = await NextDocument.getInitialProps(ctx);

      return {
        ...initialProps,
        styles: (
          <>
            {initialProps.styles}
            <style id="stitches">{getCssString()}</style>
          </>
        ),
      };
    } finally {
    }
  }

  render() {
    return (
      <Html lang="en">
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
```

Its a good idea to give it an `id="stitches"` to the style tag (will document that too) because then Stitches hydrates the styles into that existing tag

**Comment by joe-bell** [comment]
@peduarte great, thanks for the clarification 🙏🏼

**Pull request merged**